src/H5FDsubfiling/H5FDsubfiling.c

```diff
@@ -3020,20 +3311,21 @@ init_indep_io(subfiling_context_t *sf_context, int64_t file_offset, size_t io_ne
  *-------------------------------------------------------------------------
  */
 static herr_t
-iovec_fill_first(subfiling_context_t *sf_context, int64_t iovec_depth, int64_t target_datasize,
-                 int64_t start_mem_offset, int64_t start_file_offset, int64_t first_io_len,
-                 int64_t *mem_offset_out, int64_t *target_file_offset_out, int64_t *io_block_len_out)
+iovec_fill_first(subfiling_context_t *sf_context, size_t iovec_len, int64_t cur_iovec_depth,
```

Comment:
The changes in the `iovec_fill_` functions below are just changes to the buffer indexing for generating I/O vectors directly.